esp/platform/txsummary.hpp

```diff
@@ -50,80 +75,264 @@ class CTxSummary : extends CInterface
+
+    // Each entry is a subclass of TxEntryBase that allows us
+    // to keep type information along with the value to use for
+    // serialization. Each Entry has these primary attributes:
+    //
+    //  name
+    //      The name when serialized. It may be in the form
+    //      of a simplified dot-delimited JSON-style path.
+    //      That form is used to give structure when
+    //      serializing to JSON. The implementation tracks
+    //      the name in two ways- as 'name' and 'fullname'
+    //      to aid with serialization.
+    //
+    //  logLevel
+    //      The minimum logLevel that must be requested during
+    //      serialization in order for this element to appear
+    //      in the output.
+    //
+    //  group
+    //      A bit flag indicating which groups this element
+    //      belongs to. Serialization can specify which groups
+    //      are included in output
+    //
+    //  value
+    //      Subclasses of TxEntryBase include a value member of
+    //      a type appropriate for the entry.
+    //
+    struct TxEntryBase : public CInterface
     {
-        StringAttr key;
-        StringAttr value;
+        // The complete name of the entry, may be a dot-delimited
+        // path.
+        StringAttr fullname;
+
+        // The rightmost path part of fullname. During serialization
+        // to JSON an entry's fullname is tokenized and TxEntryObjects
+        // are created- one from each token of the path, with its
+        // name equal to the path token.
+        StringAttr name;
+
+        // Some Entry subclasses can assign a suffix to serialize
+        // after its value
+        StringAttr suffix;
+
+        // The minimum logLevel at which this entry is serialized
         LogLevel logLevel = LogMax;
 
-        Entry(const char* _key, const char* _value, const LogLevel _logLevel) : key(_key), value(_value), logLevel(_logLevel) {}
+        // The bit field indicating with groups this entry belongs to
+        unsigned int group = DEFAULT_GROUP;
+
+        TxEntryBase(const char* _key, const LogLevel _logLevel, const unsigned int _group = DEFAULT_GROUP, const char* _suffix = nullptr);
+        virtual bool contains(const char* path) {return false;}
+        virtual TxEntryBase* queryEntry(const char* path) {return nullptr;}
+        virtual TxEntryBase* ensurePath(const char* path) {return nullptr;}
+        virtual bool append(TxEntryBase* entry) {return false;}
+        virtual bool set(TxEntryBase* entry) {return false;}
+        virtual unsigned __int64 size() {return 1;}
+        virtual void clear() {};
+        virtual bool shouldSerialize(const LogLevel requestedLevel, const unsigned int requestedGroup);
+        virtual StringBuffer& serialize(StringBuffer& buf, const LogLevel requestedLevel, const unsigned int requestedGroup, const unsigned int requestedStyle);
+        virtual TxEntryBase* cloneWithNewName(const char* newName = nullptr){return nullptr;};
```

Comment:
If this base is never meant to be instantiated, I would make this abstract and force it to be implemented.

Reply:
Yeah makes sense.